src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/NameSyntaxClassifier.cs

```diff
@@ -321,6 +322,29 @@ private static bool IsInVarContext(NameSyntax name)
             return false;
         }
 
+        private static bool TryClassifyMethodGroupAssignment(
+            NameSyntax name,
+            SymbolInfo symbolInfo,
+            ArrayBuilder<ClassifiedSpan> result)
+        {
+            // If the syntax matches "var x = m" or "x = m", and there is a candidate symbol
+            // for m that is a method then classify it as such.
+            if (name is IdentifierNameSyntax identifierName &&
+                symbolInfo.Symbol == null &&
+                symbolInfo.CandidateSymbols.Length == 1 &&
+                symbolInfo.CandidateReason == CandidateReason.OverloadResolutionFailure &&
+                symbolInfo.CandidateSymbols[0] is IMethodSymbol methodSymbol)
+            {
+                if (identifierName.IsRightSideOfAnyAssignExpression() || identifierName.IsVariableDeclaratorValue())
```

Comment:
Why is this being limited to only assignments? Because it looks like we have a bug here where I'm passing a method as an argument too?

![image](https://user-images.githubusercontent.com/201340/139486765-a49b484e-674b-4f28-bac7-806061e24ad3.png)

Reply:
It is limited only to the bugs I knew about :)

Reply:
Also, makes me wonder if method-like (Delegate, Action, Func) identifiers should be classified as method names. Totally separate conversation from this tho.

Reply:
yeah, i agree with jason.  i would even simplify this further.  if we have a SymbolInfo with an error, and a single candidate, just extract that and classify whatever we have with that symbol.

Reply:
> if we have a SymbolInfo with an error, and a single candidate, just extract that and classify whatever we have with that symbol.

This is a little too broad I think, and breaks a bunch of tests in not good ways. [Jason's suggestion](https://github.com/dotnet/roslyn/pull/57410/files#r739463624) only breaks 2 tests, and in ways that seem reasonable to me.

Reply:
I'm curious what it breaks. Can you clarify?

Reply:
For example, the two attributes [here](https://github.com/dotnet/roslyn/blob/main/src/EditorFeatures/CSharpTest/Classification/SemanticClassifierTests.cs#L1978-L1984) are classified as classes with the change, which goes against what the comments in the test say.

Reply:
IMO.  THat seems pointless.  THe code is in error, and the compiler said: welp... the best thing i can come up with is that they bound to these two symbols (illegally).   Things like f12 on the code will then take you to the besft-effort classes.

In those cases, i see no reason why it's a problem for classification to go: yeah... your code is wrong... but it looks like you did bind to these classes, so i'll classify this as a class.

--

In other words... i view the entire endeavor to *not* classify here very suspect and somewhat of a fools errand :)  It's *extra* work to stop doing a thing that is itself not bad at all :)